OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,1142 @@
+  /// \throws OpenSim::ColumnDoesNotExist If the col label specified is not in
+  ///                                     the DataTable_.
+  SimTK::VectorView_<ET> updCol(const string& collabel) {
+    try {
+      return data.updCol(static_cast<int>(col_ind.at(collabel)));
+    } catch (std::out_of_range exc) {
+      std::string expl{"Column label '" + collabel + "' does not exist."};
+      throw ColumnDoesNotExist{expl};
+    }
+  }
+
+  /// Get an element of the DataTable_ by its index-pair(row, col). The returned
+  /// element is read-only. use updElt to get a writable reference.
+  ///
+  /// \throws SimTK::Exception::IndexOutOfRange If the row/col index specified
+  ///                                           is not in the DataTable_.
+  const ET& getElt(const size_t row, const size_t col) const {
+    return data.getElt(static_cast<int>(row), static_cast<int>(col));
+  }
+
+  /// Get an element of the DataTable_ by (row, col-label) pair. The returned
+  /// element is read-only. use updElt to get a writable reference.
+  ///
+  /// \throws SimTK::Exception::IndexOutOfRange If the row index specified
+  ///                                           is not in the DataTable_.
+  /// \throws OpenSim::ColumnDoesNotExist If the col label specified is not in
+  ///                                     the DataTable_.
+  const ET& getElt(const size_t row, const string& collabel) const {
+    try {
+      return data.getElt(static_cast<int>(row), 
+                         static_cast<int>(col_ind.at(collabel)));
+    } catch (std::out_of_range exc) {
+      std::string expl{"Column label '" + collabel + "' does not exist."};
+      throw ColumnDoesNotExist{expl};
+    }
+  }
+
+  /// Get an element of the DataTable_ by its index-pair(row, col). The returned
+  /// element is editable.
+  ///
+  /// \throws SimTK::Exception::IndexOutOfRange If the row/col index specified
+  ///                                           is not in the DataTable_.
+  ET& updElt(const size_t row, const size_t col) {
+    return data.updElt(static_cast<int>(row), static_cast<int>(col));
+  }
+
+  /// Get an element of the DataTable_ by (row, col-label) pair. The returned
+  /// element is editable.
+  ///
+  /// \throws SimTK::Exception::IndexOutOfRange If the row index specified
+  ///                                           is not in the DataTable_.
+  /// \throws OpenSim::ColumnDoesNotExist If the col label specified is not in
+  ///                                     the DataTable_.
+  ET& updElt(const size_t row, const string& collabel) {
+    try {
+      return data.updElt(static_cast<int>(row), 
+                         static_cast<int>(col_ind.at(collabel)));
+    } catch (std::out_of_range exc) {
+      std::string expl{"Column label '" + collabel + "' does not exist."};
+      throw ColumnDoesNotExist{expl};
+    }
+  }
+
+  /// Get a *copy* of the underlying matrix of the DataTable_.
+  std::unique_ptr<SimTK::Matrix_<ET>> getAsMatrix() const {
```

Comment:
Now I'm thinking that if we are handing out RowVectorViews and VectorViews, we might as well hand out a MatrixView. That would enable a lot of convenient operations. Is there a good reason not to do that since the DataTable is defined to contain a 2D matrix of elements anyway?